code/modules/power/supermatter/supermatter.dm

```diff
@@ -173,130 +233,169 @@
 
 	removed.oxygen += max((device_energy + removed.temperature - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
 
+	if (debug)
+		visible_message("[src]: Releasing [round(removed.temperature)] temp.")
+		visible_message("[src]: Releasing [round(removed.toxins)] toxins.")
+		visible_message("[src]: Releasing [round(removed.oxygen)] oxygen.")
+
+		//visible_message("[src]: Releasing additional [round((heat_capacity_new - heat_capacity)*removed.temperature)] W with exhaust gasses.")
+
 	env.merge(removed)
 
 	for(var/mob/living/carbon/human/l in view(src, min(7, round(power ** 0.25)))) // If they can see it without mesons on.  Bad on them.
 		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
 			l.hallucination = max(0, min(200, l.hallucination + power * config_hallucination_power * sqrt( 1 / max(1, get_dist(l, src)) ) ) )
 
-	for(var/mob/living/l in range(src, round((power / 100) ** 0.25)))
-		var/rads = (power / 10) * sqrt( 1 / max(get_dist(l, src),1) )
-		l.rad_act(rads)
+	for(var/mob/living/l in range(src, round((power / 100) ** 0.5)))
+		var/rads = (power / 5) * sqrt( 1 / max(get_dist(l, src),1) )
+		l.apply_effect(rads, IRRADIATE)
 
 	power -= (power/500)**3
 
+	// Lighting based on power output.
+	SetLuminosity(Clamp(round(Clamp(power/max_power,0,1)*max_luminosity), 4, max_luminosity))
+
 	return 1
 
-/obj/machinery/power/supermatter_shard
 
-/obj/machinery/power/supermatter_shard/bullet_act(obj/item/projectile/Proj)
+/obj/machinery/power/supermatter/bullet_act(var/obj/item/projectile/Proj)
```

Comment:
Dream maker spits out these errors when I try to compile after removing the var/

code\modules\power\supermatter\supermatter.dm:269:error: Proj.firer: undefined var
code\modules\power\supermatter\supermatter.dm:278:error: Proj: undefined var
code\modules\power\supermatter\supermatter.dm:279:error: Proj.damage: undefined var
code\modules\power\supermatter\supermatter.dm:281:error: Proj.damage: undefined var